programs/psa/crypto_examples.c

```diff
@@ -1,23 +1,15 @@
 #include "psa/crypto.h"
 #include <string.h>
-
-#if defined(MBEDTLS_PLATFORM_C)
-#include "mbedtls/platform.h"
```

Comment:
All example programs include `mbedtls/platform.h` and use indirections like `mbedtls_printf` and `MBEDTLS_EXIT_FAILURE`. Why did you remove these indirections, but only in `programs/psa/*.c` and not in other example/utility programs?

Reply:
The plan is to make platform.h an internal header when we can. PSA programs can set a good example and avoid using platform.h. Other programs could remove platform.h as well, but they are still Mbed TLS examples and use Mbed TLS APIs which we also want to make internal when we can.

Reply:
I thought the point of these abstractions was that if you port the library to a system where they're needed, you also get the example programs. Why would this no longer be a desirable property?

Note that I can see why we'd want PSA demonstration programs to be free of `mbedtls` dependencies. But that's out of scope of this PR.

Reply:
The property is desirable, but the cost of maintaining the abstraction is not. There are ways to use the linker to replace e.g. malloc with a custom replacement (gcc has `-Wl,-wrap,malloc`, armcc, msvc, and IAR have other mechanisms). We don't need to maintain any macros to support this.

Reply:
I agree that removing the platform module and our use of it externally is outside the scope of this PR. We can do that in a separate PR.